core/src/main/java/org/elasticsearch/index/shard/TranslogRecoveryPerformer.java

```diff
@@ -159,6 +159,9 @@ private void performRecoveryOperation(Engine engine, Translog.Operation operatio
                         logger.trace("[translog] recover [index] op of [{}][{}]", index.type(), index.id());
                     }
                     index(engine, engineIndex);
+                    if (engineIndex.getFailure() != null) {
+                        throw engineIndex.getFailure();
```

Comment:
I think we can return the engine.operation here and then check on the caller side if there were errors we care about. I think we won't care about most of them.